### ISSUE TITLE
Issue with transient fields

### ISSUE THREAD

**Issue by kstokoz** [plokhotnyuk/jsoniter-scala#459]
Existing test for transient functionality:

```
"don't serialize and deserialize transient and non constructor defined fields of case classes" in {
      case class Transient(@transient transient: String = "default", required: String) {
        val ignored: String = s"$required-$transient"
      }

      val codecOfTransient = make[Transient](CodecMakerConfig)
      verifySer(codecOfTransient, Transient(required = "VVV"), """{"required":"VVV"}""")
      verifyDeser(codecOfTransient, Transient(required = "VVV"), """{"transient":"XXX","required":"VVV"}""")
    }
```

probably works due to `transientDefault` flag being enabled by default.
Following case failing for me, even though i would expect it to pass:
```
      verifySer(codecOfTransient, Transient(required = "VVV", transient = "non-default-value"), """{"required":"VVV"}""")
```

**Comment by plokhotnyuk** [comment]
Hi, Kyrylo!

I've checked the macro implementation and [added missing test cases](https://github.com/plokhotnyuk/jsoniter-scala/commit/31b205114d93411874176c80b0c5a20c4bf1dd17) - they pass for me.

Please, double check on your side if you have used the proper annotation (should be `com.github.plokhotnyuk.jsoniter_scala.macros.transient`). 

If you will still have unexpected behavior please create a PR which reproduce it.

**Comment by kstokoz** [comment]
Thanks for looking into it. 

Indeed after recompiling the project all works fine, i guess there is issues with compiled classes caching when i add/remove annotation sometimes serialized output not in sync with source code.

**Comment by domdorn** [comment]
sorry for commenting on an old issue.. could we have the same behaviour of `com.github.plokhotnyuk.jsoniter_scala.macros.transient` for fields that are simply annotated with `scala.transient` ? This would have been the expected behaviour for me.. only after this didn't work and I was trying for other ways to implement this behaviour I found this issue.

**Comment by plokhotnyuk** [comment]
@domdorn Hi, Dominik!

Thanks for reaching me here! 

Do not hesitate to open a new issue with a reference to old one.

I can add support of `scala.transient` using an additional compile-time option, that will be turned off by default to preserve current behavior for existing usage.

So you will be able use it with code like here:
`JsonCodecMaker.make(CodecMakerConfiguration.withScalaTransientSupport(true))`

Please let me know if it will be acceptable for your case and I will proceed with implementation and cutting a release for you.
